core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -277,10 +277,30 @@ public void checkCompatibility(MappedFieldType other, List<String> conflicts) {
         if (!names().equals(other.names())) {
             conflicts.add("mapper [" + names().fullName() + "] has different index_name");
         }
-
         if (Objects.equals(similarity(), other.similarity()) == false) {
             conflicts.add("mapper [" + names().fullName() + "] has different similarity");
         }
+
+        if (strict) {
+            if (omitNorms() != other.omitNorms()) {
```

Comment:
I think norms are complicated because disabling them is ok while enabling them will not work?